### DISCUSSION TITLE
TaskGraph naming

### DISCUSSION THREAD

**Discussion by andrii0lomakin** [beehive-lab/TornadoVM#504] [Q&A] [answered]
Good day.

Why is TaskGraph called so instead of TaskQueue? Seems this class plays the role of a command queue.
I am asking because I suspect I missed some functions that make it graph instead of queue.

**Comment by jjfumero** [comment, marked as answer]
Hi @andrii0lomakin . `TaskGraph` is a much higher abstraction compared to command queues. We can think about command queues as implementation details. The idea behind task-graphs is that task within the graph can be scheduled in different order and different devices by analysing the data dependencies across tasks. In fact, TornadoVM can use different command queues per task. 

By default, TornadoVM takes care of scheduling. However, developers can control where to execute and parameters for execution using the execution plan.

**Reply by andrii0lomakin** [reply]
Hi @jjfumero 
Got it. Thank you for the clarification.
